### PR TITLE
make streaming from pending deploys work

### PR DESCRIPTION
**Pull request by grosser**
by waiting until the execution is ready

fixes fallout from https://github.com/zendesk/samson/pull/3013
which declared the job done when it ran on a pending deploy

@zendesk/bre @adammw

### CODE REVIEW

File: app/channels/job_outputs_channel.rb [hunk 1]
```diff
@@ -50,27 +52,28 @@ def status_response(event)
...
-      job = Job.find(id)
-      builder = EventBuilder.new(job)
-      transmit event: :started, data: builder.payload(:started, nil)
-      job.output.each_line do |line|
-        transmit event: :message, data: builder.payload(:message, line)
-      end
-      transmit event: :finished, data: builder.payload(:finished, nil)
     end
+
+    # TODO: kill threads when user stops streaming
```
Comment (grosser) [inline app/channels/job_outputs_channel.rb]: got a branch for that ready `grosser/start-2`but it's kinda tricky since it would cancel streams when user opens multiple tabs and might have a race condition if user reloads and sends the disconnect too late .. so not doing that for now

Comment (grosser) [reply app/channels/job_outputs_channel.rb]: kk, fixed :)

File: app/helpers/deploys_helper.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ module DeploysHelper
   }.freeze
 
   def deploy_output
-    output = ActiveSupport::SafeBuffer.new
+    output = "".html_safe
```
Comment (adammw) [inline app/helpers/deploys_helper.rb]: why the change?

File: app/channels/job_outputs_channel.rb [hunk 3]
```diff
@@ -50,33 +53,49 @@ def status_response(event)
...
+          execution = JobQueue.find_by_id(id)
+          break stream_execution(execution) if execution
+
+          # cancelled or we missed it running
+          job = Job.find(id)
+          break stream_finished_output(job) if job.finished?
+
+          # pending
+          sleep WAIT_DURATION
         end
```
Comment (adammw) [inline app/channels/job_outputs_channel.rb]: does it close the connection if it gets here (times out waiting)?

File: app/helpers/deploys_helper.rb (hunk 2 above)
Comment (grosser) [reply app/helpers/deploys_helper.rb]: just readability

File: app/channels/job_outputs_channel.rb (hunk 3 above)
Comment (grosser) [reply app/channels/job_outputs_channel.rb]: the channel does not get closed automatically, but since it's not doing anything it does not really matter (same as before)

Comment (adammw) [reply app/channels/job_outputs_channel.rb]: seems like it should get closed if it gets to here, unless there's a reason not to (like it'll reconnect or something)

Comment (grosser) [reply app/channels/job_outputs_channel.rb]: if we do a `connection.close` then the client reconnects and streams the whole output again :)
... there might be a way to make this work, but it involves a lot of digging/checking for little gain (closing connections that do nothing, not sure if it even removes the channel and which are auto-closed when user closes the tab)
